AuthServ/AuthServer.cpp

```diff
@@ -93,10 +93,34 @@ void auth_init(AuthServer_Private& client)
     /* BCast Shard Capabilities */
     START_REPLY(e_AuthToCli_ServerCaps);
     uint32_t bufSzPos = client.m_buffer.tell();
+#ifdef DS_OU_COMPATIBLE
+    // OpenUru parses this as the start of a FileDownloadChunk message,
+    // which will be ignored,
+    // because no file download transaction is active.
+    client.m_buffer.write<uint32_t>(37);
+
+    // H'uru sees: 0x25 bytes of data, bit vector is 1 dword long, value is [caps]
+    // OpenUru sees: message type 0x25 (FileDownloadChunk), transaction ID 0x10000, error code 0x[caps]0000, file size 0x0000[caps] (continued below)
+    caps.write(&client.m_buffer);
+#ifdef DEBUG
+    uint32_t newBufPos = client.m_buffer.tell();
+    if ((newBufPos - bufSzPos) > (3 * sizeof(uint32_t)))
+        ST::printf(stderr, "[Auth] ServerCaps compatibility broken by size change");
```

Comment:
IMO this should be a hard error, not just a print and continuing normally. If the size is mismatched, the message is guaranteed to not be parseable correctly.

Perhaps this check should always be enabled, not just in debug mode.

Reply:
I can probably change this to a `DS_ASSERT` that would throw if the condition isn't met